src/main/java/randoop/reflection/TypeInstantiator.java

```diff
@@ -98,6 +99,140 @@ public TypedClassOperation instantiate(TypedClassOperation operation) {
     return operation;
   }
 
+  /**
+   * Creates a set of instantiations of a given operation that is generic or has wildcard types.
+   *
+   * @param operation an operation that is generic or has wildcard types
+   * @return all possible instantiations of the given operation
+   */
+  public Set<TypedClassOperation> instantiateWithMultipleTypes(TypedClassOperation operation) {
+    assert operation.isGeneric() || operation.hasWildcardTypes()
+        : "operation must be generic or have wildcards";
+
+    // This list is the same size as the set returned by this method.
+    List<Substitution> substitutions = new ArrayList<>();
+
+    ClassOrInterfaceType declaringType = operation.getDeclaringType();
+    // If the declaring type of the operation is generic, get the set of possible instantiations.
+    if (declaringType.isGeneric()) {
+
+      // If the operation creates objects of its declaring type, it may create a new instantiation.
+      if (operation.isConstructorCall()
+          || (operation.isStatic() && operation.getOutputType().equals(declaringType))) {
+        if (declaringType.isSubtypeOf(JDKTypes.SORTED_SET_TYPE)) {
+          instantiationsForSortedSetType(operation, substitutions);
+        } else {
+          instantiationsForGeneralTypes(declaringType, substitutions);
+        }
+      } else {
+        // Otherwise, select from existing ones.
+        for (Type type : inputTypes) {
+          if (type.isParameterized()
+              && ((InstantiatedType) type).isInstantiationOf(declaringType)) {
+            substitutions.add(
+                ((InstantiatedType) type).getInstantiatingSubstitution(declaringType));
+          }
+        }
+      }
+    }
+
+    Set<TypedClassOperation> result = new HashSet<>();
+    // Apply each one of these type substitutions and create an instantiated operation.
+    for (Substitution substitution : substitutions) {
```

Comment:
If the declaring type of the operation is *not* generic, then this loop is never entered and wildcard types are not capture-converted.  That seems problematic.

Reply:
I think the loop is entered, but substitutions will be empty so nothing is done.  I agree this does not sound right.  The 'instantiate' method (directly above) does applyCaptureConversion for wildcard types.  Should this method do the same thing?